operator/pkg/ciliumidentity/cache.go

```diff
@@ -162,31 +169,20 @@ func (c *CIDUsageInPods) AssignCIDToPod(podName, cidName string) (string, int) {
 
 // RemovePod removes the pod from the pod to CID map, decrements the CID usage
 // and returns the CID name and its usage count after decrementing the usage.
+// If the pod does not exist in the pod to CID map it returns empty values.
 // The return values are used to track when old CIDs are no longer used.
-func (c *CIDUsageInPods) RemovePod(podName string) (string, int, error) {
+func (c *CIDUsageInPods) RemovePod(podName string) (string, int, bool) {
 	c.mu.Lock()
 	defer c.mu.Unlock()
 
 	cidName, exists := c.podToCID[podName]
 	if !exists {
-		return "", 0, errors.New("cilium identity not found in pods usage cache")
+		return "", 0, false
```

Comment:
What is the benefit of returning a bool instead of an error, here?

Reply:
Would you mind doing that in a separate commit?
I think that all the changes to the `CIDUsageInPods` type (adding the logger, changing this return value and removing the unused `CIDUsedByPod`) should be all part of their own commit, since they have not much to do with the time trackers.

Reply:
Extracted